easybuild/tools/systemtools.py

```diff
@@ -998,6 +1008,123 @@ def get_glibc_version():
     return glibc_ver
 
 
+def get_cuda_object_dump_raw(path):
+    """
+    Get raw ouput from command which extracts information from CUDA binary files in a human-readable format,
+    or None for files containing no CUDA device code.
+    See https://docs.nvidia.com/cuda/cuda-binary-utilities/index.html#cuobjdump
+    """
+
+    res = run_shell_cmd("file %s" % path, fail_on_error=False, hidden=True, output_file=False, stream_output=False)
+    if res.exit_code != EasyBuildExit.SUCCESS:
+        fail_msg = "Failed to run 'file %s': %s" % (path, res.output)
+        _log.warning(fail_msg)
+
+    # check that the file is an executable or library/object
+    result = None
+    if any(x in res.output for x in ['executable', 'object', 'library']):
+        cuda_cmd = f"cuobjdump {path}"
```

Comment:
What if `cuobjdump` is not around? At least emit a clear log message.

Maybe also change return type of this function:

* `(None, f"{path} is not an executable, object, library")`
* `(None, f"{path} does not contain CUDA device code")`
* `(False, "cuobjdump command not found")`
* `(False, f"cuobjdump failed: {res.output}")`
* `(True, res.output)`

Reply:
`cuobjdump` is part of the CUDA toolkit, so it really should be found; it should be an error if not

Reply:
The case I am worried about is EESSI as in there we demote `CUDA` to a build dependency (and rely on rpath linking to access the runtime). Right now, it seems the build dependency is still loaded during the sanity check step so I think we are covered...but there is a danger that this behaviour will change.

Reply:
I think it doesn't hurt to check for the presence of `cuobjdump`. The check for the CUDA toolkit happens in the sanity check, but in the future, _someone else_ might call this function from a different place and have no such check for CUDA toolkit in place.

Reply:
I guess you could work around it by adding an EB configuration option to point to a particular `cuobjdump` to use for the sanity check? e.g. `--cuda-sanity-check-cuobjdump=/path/to/some/cuobjdump`. We don't need the CUDA toolkit version that provides that `cuobjdump` to match the version used in the installation

Reply:
First, let's see if it's actually an issue, as with `EESSI-extend`, I haven't seen it (even though, like Alan, I did expect it)... We can easily fix that in a subsequent PR